dgl_sparse/src/matrix_ops.cc

```diff
@@ -58,12 +55,112 @@ std::tuple<std::shared_ptr<COO>, torch::Tensor, torch::Tensor> COOIntersection(
   return {ret_coo, lhs_indices, rhs_indices};
 }
 
+torch::Tensor RevertIndices(const torch::Tensor& indices) {
+  auto rev_tensor = torch::empty_like(indices);
+  rev_tensor.index_put_(
+      {indices}, torch::arange(0, indices.numel(), rev_tensor.options()));
+  return rev_tensor;
+}
+
+std::tuple<torch::Tensor, torch::Tensor> CompactIndices(
+    const torch::Tensor& row,
+    const torch::optional<torch::Tensor>& leading_indices) {
+  torch::Tensor sorted, sort_indices, uniqued, unique_reverse_indices, counts;
+  int64_t n_leading_indices = 0;
```

Comment:
```suggestion
  // 1. Sort leading indices and row indices in ascending order.
  int64_t n_leading_indices = 0;
```
Add comments for each step for better readability.